orte/mca/iof/orted/iof_orted.c

```diff
@@ -297,13 +306,17 @@ static int orted_close(const orte_process_name_t* peer,
                 }
                 ++cnt;
             }
+#if OPAL_PMIX_V1
             if (ORTE_IOF_STDDIAG & source_tag) {
                 if (NULL != proct->revstddiag) {
                     orte_iof_base_static_dump_output(proct->revstddiag);
                     OBJ_RELEASE(proct->revstddiag);
                 }
                 ++cnt;
             }
+#else
+            ++cnt;
```

Comment:
note to myself, did the recent changes impact this (e.g. do we still __always__ need to wait for `4 == cnt` in order to remove `proct` ?

Reply:
I think there might be a problem when stdout and stderr are merged as you don't iof.push the stderr in that case. This means there won't be a stderr entry for that proc, and so it revstderr will be NULL. However, so long as you iof.close stderr, the count will still be correct. I'd therefore suggest checking that you always call iof.close on stderr, or we'll need to adjust the code to increment cnt accordingly.

Reply:
I got rid of the count and replaced it with a more robust solution - see what you think